### PR TITLE
Produce more sensible matcher names

### PR DESCRIPTION
**Pull request by zliu41**
Nil_match becomes List_match

### CODE REVIEW

File: plutus-tx-plugin/plutus-tx-plugin.cabal [hunk 1]
```diff
@@ -121,10 +121,6 @@ test-suite plutus-tx-tests
   if flag(use-ghc-stub)
     buildable: False
 
-  -- test output changes after 9.2, bad for golden tests
-  if impl(ghc >=9.3)
-    buildable: False
```
Comment (effectfully) [inline plutus-tx-plugin/plutus-tx-plugin.cabal:126]: Is this a related change?

**Review by effectfully — APPROVED** [review]
Much better, thanks.

File: plutus-tx-plugin/plutus-tx-plugin.cabal (hunk 1 above)
Comment (zliu41) [reply plutus-tx-plugin/plutus-tx-plugin.cabal:126]: No but this was committed by accident in https://github.com/input-output-hk/plutus/pull/5528 and I need to remove it to run the tests for 9.6.

**Review by michaelpj — COMMENTED** [review]
Good catch!
